virtualmachine/vsphere/vm.go

```diff
@@ -1775,20 +1788,37 @@ func GetVmList(vm *VM, markedTemplate bool) ([]map[string]interface{}, error) {
 			!markedTemplate && vmo.Config.Template {
 			continue
 		}
-		vminfo := getVmInfo(vmo, name)
+		vminfo := getVmInfo(vmo, vmProp.Name)
 		vmList = append(vmList, vminfo)
 	}
 	return vmList, nil
 }
 
-// getVirtualMachines : Returns the virtual machines in a dc/cluster/host
-func getVirtualMachines(vm *VM) (map[string]mo.VirtualMachine, error) {
+// getVirtualMachines : Returns the virtual machines in a allDCs/dc/cluster/host
+func getVirtualMachines(vm *VM, allDCs bool) ([]VmProperties, error) {
```

Comment:
can we move this func to util.go file ? please break this big func to smaller ones.

Reply:
Moved all related functions to utils.go and did split getVirtualMachines into smaller functions.